lua/wikis/commons/Widget/Infobox/UpcomingTournaments/Row.lua

```diff
@@ -0,0 +1,133 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/Infobox/UpcomingTournaments/Row
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Lua = require('Module:Lua')
+
+local Array = Lua.import('Module:Array')
+local Class = Lua.import('Module:Class')
+local Countdown = Lua.import('Module:Countdown')
+local DateExt = Lua.import('Module:Date/Ext')
+local FnUtil = Lua.import('Module:FnUtil')
+local HighlightConditions = Lua.import('Module:HighlightConditions')
+local LeagueIcon = Lua.import('Module:LeagueIcon')
+
+local Widget = Lua.import('Module:Widget')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Div = HtmlWidgets.Div
+local Link = Lua.import('Module:Widget/Basic/Link')
+local Span = HtmlWidgets.Span
+local Tr = HtmlWidgets.Tr
+local Td = HtmlWidgets.Td
+
+---@class UpcomingTournamentsRowParameters
+---@field data placement
+---@field onlyHighlightOnValue string?
+
+---@class UpcomingTournamentsRow: Widget
+---@operator call(UpcomingTournamentsRowParameters): UpcomingTournamentsRow
+---@field props UpcomingTournamentsRowParameters
+local UpcomingTournamentsRow = Class.new(Widget)
+
+---@return Widget?
+function UpcomingTournamentsRow:render()
+	local data = self.props.data
+	if not data then return end
+	local status = (data.extradata or {}).status
+	if status == 'cancelled' or status == 'delayed' then return end
+
+	return HtmlWidgets.Table{
+		classes = {'wikitable', 'wikitable-striped', 'infobox_matches_content'},
+		children = {
+			Tr{
+				children = Td{
+					classes = Array.extend(
+						'versus',
+						self:_isHighlighted() and 'tournament-highlighted-bg' or nil
+					),
+					css = {['text-align'] = 'center'},
+					children = {
+						LeagueIcon.display{
+							icon = data.icon,
+							iconDark = data.icondark,
+							link = data.pagename
+						},
+						' ',
+						Link{
+							link = data.pagename,
+							children = data.tournament
+						}
+					}
+				}
+			},
+			Tr{
+				children = Td{
+					classes = {'match-filler'},
+					children = {
+						self:_getCountdown(),
+						Div{
+							css = {
+								width = '100px',
+								float = 'right',
+								['white-space'] = 'nowrap'
+							},
+							children = Div{
+								classes = {'tournament-span'},
+								children = self:_getTournamentSpan()
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+end
+
+---@private
+---@return boolean
+function UpcomingTournamentsRow:_isHighlighted()
+	return HighlightConditions.tournament(self.props.data, {onlyHighlightOnValue = self.props.onlyHighlightOnValue})
```

Comment:
do we need the `onlyHighlightOnValue` stuff?
like iirc it is only used in tournament lists on some series pages and the likes
not sure it should be applicable for team/player pages

maybe @iMarbot can share his opinion (iirc cs is one of the wikis actively using that option)

Reply:
> edit: I am starting to hate local overrides :(

at least it is tracked on git^^

Reply:
> edit: I am starting to hate local overrides :(

that's why I'm trying to minimize them :D